src/FSharp.Core/seq.fsi

```diff
@@ -2935,3 +2935,289 @@ module Seq =
+    /// <exception cref="T:System.ArgumentException">Thrown when the input sequence is empty.</exception>
+    /// <exception cref="T:System.ArgumentException">Thrown when count is less than 0.</exception>
+    ///
+    /// <example id="randomChoicesWith-1">
+    /// <code lang="fsharp">
+    /// let inputs = seq { 0; 1; 2; 3; 4 }
+    ///
+    /// inputs |> Seq.randomChoicesWith Random.Shared 3
+    /// </code>
+    /// Can evaluate to <c>seq { 3; 1; 3 }</c>.
+    /// </example>
+    [<CompiledName("RandomChoicesWith")>]
+    val randomChoicesWith: random: Random -> count: int -> source: seq<'T> -> seq<'T>
+
+    /// <summary>
+    /// Returns a sequence of random elements from the given sequence with the specified <c>randomizer</c> function, each element can be selected multiple times.
+    /// </summary>
+    ///
+    /// <param name="randomizer">The randomizer function, must return a float number from [0.0..1.0) range.</param>
+    /// <param name="count">The number of elements to return.</param>
+    /// <param name="source">The input sequence.</param>
+    ///
+    /// <returns>A sequence of randomly selected elements from the input sequence.</returns>
+    ///
+    /// <exception cref="T:System.ArgumentNullException">Thrown when the input sequence is null.</exception>
+    /// <exception cref="T:System.ArgumentException">Thrown when the input sequence is empty.</exception>
+    /// <exception cref="T:System.ArgumentException">Thrown when count is less than 0.</exception>
+    /// <exception cref="T:System.ArgumentOutOfRangeException">Thrown when the randomizer function returns a number outside the range [0.0..1.0).</exception>
+    ///
+    /// <example id="randomChoicesBy-1">
+    /// <code lang="fsharp">
+    /// let inputs = seq { 0; 1; 2; 3; 4 }
+    ///
+    /// inputs |> Seq.randomChoicesBy Random.Shared.NextDouble 3
+    /// </code>
+    /// Can evaluate to <c>seq { 3; 1; 3 }</c>.
+    /// </example>
+    [<CompiledName("RandomChoicesBy")>]
+    val randomChoicesBy: randomizer: (unit -> float) -> count: int -> source: seq<'T> -> seq<'T>
+
+    /// <summary>
+    /// Returns a random sample of elements from the given sequence, each element can be selected only once.
+    /// </summary>
+    ///
+    /// <param name="count">The number of elements to return.</param>
+    /// <param name="source">The input sequence.</param>
+    ///
+    /// <returns>A sequence of randomly selected elements from the input sequence.</returns>
+    ///
+    /// <exception cref="T:System.ArgumentNullException">Thrown when the input sequence is null.</exception>
+    /// <exception cref="T:System.ArgumentException">Thrown when the input sequence is empty.</exception>
+    /// <exception cref="T:System.ArgumentException">Thrown when count is less than 0.</exception>
+    /// <exception cref="T:System.ArgumentOutOfRangeException">Thrown when count is greater than the length of the input sequence.</exception>
```

Comment:
Note that the tests are correct and the code is not. This helped me discover a [bug](https://github.com/dotnet/fsharp/pull/17326) in our exception checking.

Reply:
@psfinaki I think code is also correct, since `invalidArgOutOfRange` really throws `ArgumentException`, so I'm only updating the description
```fsharp
    let invalidArgOutOfRange (arg:string) (index:int) (text:string) (bound:int) =
        invalidArgFmt arg
            "{0}\n{1} = {2}, {3} = {4}"
            [|SR.GetString SR.outOfRange; arg; index; text; bound|]
```

Reply:
So @Lanayx this doesn't hold anymore, we've just merged the fix that makes this function throw the right exception. Hence the code here needs to adjust :)

Reply:
I'm not sure what is "right exception" anymore :(

skip throws InvalidArgOutOfRange when count > n
take throws InvalidOperation when count > n

if sample will throws InvalidArg, it will be 3 different exception thrown in similar situations. I suggest we throw InvalidArgOutOfRange in this case as well (similar to skip) and update RFC

Reply:
By the way, just noticed, that I indeed wanted OutOfRange in this case :) https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1135-random-functions-for-collections.md - the link is correct, just the alias is not

Reply:
Heh that's a good observation and a good intention as well. My feeling is that this would be quite a big effort: RFCs, discussions, docs, breaking changes - you know, code and tests will be the quickest and easiest things here maybe :)

I'd suggest we implement things here according to the current RFC and then start a discussion about the unification of these exceptions. That might be a long and somewhat annoying process, I don't think we want to block this PR by it.

Reply:
Done

Reply:
Thanks!